includes/class-wcpdf-admin.php

```diff
@@ -429,45 +427,51 @@ public function output_number_date_edit_fields( $document, $data ) {
 				<!-- Title -->
 				<h4>
 					<?= $document->get_title(); ?>
-					<span class="wpo-wcpdf-edit-date-number dashicons dashicons-edit"></span>
-					<span class="wpo-wcpdf-delete-document dashicons dashicons-trash" data-nonce="<?php echo wp_create_nonce( "wpo_wcpdf_delete_document" ); ?>"></span>
-					<?php do_action( 'wpo_wcpdf_document_actions', $document ); ?>
+					<?php if( $document->exists() ) : ?>
+						<span class="wpo-wcpdf-edit-date-number dashicons dashicons-edit"></span>
+						<span class="wpo-wcpdf-delete-document dashicons dashicons-trash" data-nonce="<?php echo wp_create_nonce( "wpo_wcpdf_delete_document" ); ?>"></span>
+						<?php do_action( 'wpo_wcpdf_document_actions', $document ); ?>
+					<?php endif; ?>
 				</h4>
 
 				<!-- Read only -->
 				<div class="read-only">
-					<div class="<?= $document->get_type(); ?>-number">
-						<p class="form-field <?= $data['number']['name']; ?>_field">	
-							<p>
-								<span><strong><?= $data['number']['label']; ?></strong></span>
-								<span><?= $data['number']['formatted']; ?></span>
+					<?php if( $document->exists() ) : ?>
+						<div class="<?= $document->get_type(); ?>-number">
+							<p class="form-field <?= $data['number']['name']; ?>_field">	
+								<p>
+									<span><strong><?= $data['number']['label']; ?></strong></span>
+									<span><?= $data['number']['formatted']; ?></span>
+								</p>
 							</p>
-						</p>
-					</div>
-					<div class="<?= $document->get_type(); ?>-date">
-						<p class="form-field form-field-wide">
-							<p>
-								<span><strong><?= $data['date']['label']; ?></strong></span>
-								<span><?= $data['date']['formatted']; ?></span>
+						</div>
+						<div class="<?= $document->get_type(); ?>-date">
+							<p class="form-field form-field-wide">
+								<p>
+									<span><strong><?= $data['date']['label']; ?></strong></span>
+									<span><?= $data['date']['formatted']; ?></span>
+								</p>
 							</p>
-						</p>
-					</div>
-					<?php do_action( 'wpo_wcpdf_meta_box_after_document_data', $document, $document->order ); ?>
+						</div>
+						<?php do_action( 'wpo_wcpdf_meta_box_after_document_data', $document, $document->order ); ?>
+					<?php else : ?>
+						<span class="wpo-wcpdf-set-date-number button"><?php _e( 'Set invoice number & date', 'woocommerce-pdf-invoices-packing-slips' ); ?></span>
+					<?php endif; ?>
 				</div>
 
 				<!-- Editable -->
 				<div class="editable">
 					<p class="form-field <?= $data['number']['name']; ?>_field ">
 						<label for="<?= $data['number']['name']; ?>"><?= $data['number']['label']; ?></label>
-						<?php if ( ! empty( $data['number']['plain'] ) ) : ?>
+						<?php if ( $document->exists() && ! empty( $data['number']['plain'] ) ) : ?>
```

Comment:
This one, as well as the next one can actually be done inside the `get_current_values_for_document()` method (except notes, which you can fetch also for non-existing).